Express app.use argument does not match parameters

- [x] I tried using the latest `express-serve-static-core/express-serve-static-core.d.ts` file in this repo and had problems.
- [x] I tried using the latest stable version of tsc. https://www.npmjs.com/package/typescript
- [x] I have a question that is inappropriate for [StackOverflow](https://stackoverflow.com/).  (Please ask any appropriate questions there).
- [x] I want to talk about `express-serve-static-core/express-serve-static-core.d.ts`.
  - The authors of that type definition are cc/ @borisyankov 

**Problem:** app.use() with normal Express arguments in it throws the IDE error "argument does not match parameters".

Example: 

``` typescript
class Server {
  public app: express.Application;

  public static bootstrap(): Server {
    return new Server();
  }

  constructor() {
    this.app = express();

    this.config();

    this.routes();
  }

  private config() {
    //configure hbs
    this.app.set("views", path.join(__dirname, "views"));
    this.app.set("view engine", "hbs");

    //mount json form parser
    this.app.use(bodyParser.json()); //IDE error

    //mount query string parser
    this.app.use(bodyParser.urlencoded({ extended: true })); //IDE error

    //add static paths
    this.app.use(express.static(path.join(__dirname, "public"))); // IDE error

    // catch 404 and forward to error handler
    this.app.use(function(err: any, req: express.Request, res: express.Response, next: express.NextFunction) {
      var error = new Error("Not Found");
      err.status = 404;
      next(err);
    }); // IDE error
  }

  private routes() {
    //get router
    let router: express.Router;
    router = express.Router();

    //create routes
    var index: indexRoute.Index = new indexRoute.Index();

    //home page
    router.get("/", index.index.bind(index.index));

    //use router middleware
    this.app.use(router); //IDE error
  }
}
```

A workaround is to pass `this.app` to `this.config()` and `this.routers()`

My packages:

``` json
"dependencies": {
    "@types/body-parser": "0.0.33",
    "@types/express": "^4.0.33",
    "@types/express-serve-static-core": "^4.0.37",
    "@types/mime": "0.0.29",
    "@types/node": "^6.0.45",
    "@types/serve-static": "^1.7.31",
    "body-parser": "^1.15.2",
    "express": "^4.14.0",
    "express-serve-static-core": "^0.1.1",
    "handlebars": "^4.0.5",
    "hbs": "^4.0.1",
    "ts-node": "^1.6.0",
    "tslint": "^3.15.1",
    "typescript": "^2.0.3"
  }
```

I'm using IntelliJ IDEA ultimate 2016.2.5 with the Typescript interpreter from my node_modules folder.

**Update:**
When running it with ts-node, it will work
When compiling it using tsc, it will throw an error: "node_modules/@types/express-serve-static-core/index.d.ts(6,1): error TS6053: File '<truncated>/node_modules/@types/node/node.d.ts' not found.", however changing this to node/index.d.ts did not fix the issue.

--- a/express-serve-static-core/index.d.ts
+++ b/express-serve-static-core/index.d.ts
@@ -856,6 +856,10 @@ interface RequestParamHandler {
     (req: Request, res: Response, next: NextFunction, value: any, name: string): any;
 }
 
+type ApplicaitonRequestHandler<T> =  IRouterHandler<T> & IRouterMatcher<T> & {
+  (...handlers: RequestHandlerParams[]): T;
+};
+
 interface Application extends IRouter, Express.Application {
     /**
      * Express instance itself is a request handler, which could be invoked without
@@ -1096,6 +1100,8 @@ interface Application extends IRouter, Express.Application {
      * Used to get all registered routes in Express Application
      */
     _router: any;
+
+    use: ApplicaitonRequestHandler<this>;
 }
 
 interface Express extends Application {

